src/agent.rs

```diff
@@ -115,6 +115,11 @@ impl From<i32> for TLSVersion {
 /// An [`Agent`] holds a non-cloneable reference to a Stream.
 pub struct Agent {
     pub descriptor: AgentDescriptor,
+
+    #[cfg(feature = "wolfssl")]
+    pub stream: WolfSSL,
+
+    #[cfg(feature = "openssl")]
     pub stream: OpenSSL,
```

Comment:
I just thought about this problem. Maybe one way to make this cleaner would be to make `Agent` a generic `Agent<S>`, where `S: StreamKind`. The trait `StreamKind` now defines several methods like `fn reset(&self)`.

Then, we could compile-conditional `impl StreamKind for WolfSSL` or `impl StreamKind for OpenSSL`.
That way the `Agent` is decoupled from the stream kind and already generic enough for any stream kind.